src/Sema.zig

```diff
@@ -29781,10 +29781,24 @@ fn semaStructFields(mod: *Module, struct_obj: *Module.Struct) CompileError!void
                 extra_index += body.len;
                 const init = try sema.resolveBody(&block_scope, body, struct_obj.zir_index);
                 const field = &struct_obj.fields.values()[i];
-                const tree = try sema.getAstTree(&block_scope);
-                const init_src = containerFieldInitSrcLoc(decl, tree.*, 0, i);
-                const coerced = try sema.coerce(&block_scope, field.ty, init, init_src);
-                const default_val = (try sema.resolveMaybeUndefVal(&block_scope, init_src, coerced)) orelse
+                const coerced = sema.coerce(&block_scope, field.ty, init, .unneeded) catch |err| switch (err) {
+                    error.NeededSourceLocation => {
+                        const tree = try sema.getAstTree(&block_scope);
+                        const init_src = containerFieldInitSrcLoc(decl, tree.*, 0, i);
+                        _ = try sema.coerce(&block_scope, field.ty, init, init_src);
+                        return error.AnalysisFail;
+                    },
+                    else => |e| return e,
+                };
+                const default_val = sema.resolveMaybeUndefVal(&block_scope, .unneeded, coerced) catch |err| switch (err) {
+                    error.NeededSourceLocation => {
```

Comment:
Doing this check for `resolveMaybeUndefVal` isn't very useful but it should be done before the call to `failWithNeededComptime`.

Reply:
Would you prefer I wait a bit until #13532 gets merged, and then fix this and rebase?

Reply:
There is going to be merge conflicts either way so it doesn't really matter though I'll probably want Andrews opinion on that change so this might be able to get merged first.